src/components/AccessForm.vue

```diff
@@ -73,24 +79,66 @@ export default defineComponent({
             trigger: "change"
           }
         ]
-      }
+      },
+      loading: false
     };
   },
+  computed: {
+    ...mapState(["ourSecretKey"]),
+    ...mapGetters(["enclavePublicKey"])
+  },
   methods: {
+    ...mapActions(["sealBox"]),
     onSubmit(formName: string) {
       (this.$refs[formName] as typeof elForm).validate(
         async (valid: boolean) => {
           if (valid) {
-            const enclavePublicKey = this.$store.getters.enclavePublicKey;
+            this.loading = true;
             // Get the unproxied form data before encrypting, for clarity.
             const data = Object.assign({}, this.form);
-            await encryptJson(data, enclavePublicKey);
+            await this.sealBox(data).then(async messageData => {
+              if (messageData) {
+                await this.postEncryptedBox(messageData);
+              }
+            });
           } else {
             console.log("error submit!!");
             return false;
           }
         }
       );
+    },
+    async postEncryptedBox(messageData: MessageData) {
```

Comment:
I think it makes sense to move this into vuex actions rather than doing the calls in the components (like we are doing with the file upload).

Reply:
I didn't put it in vuex since it's not a reusable call and the result isn't saved.
But it makes sense to centralize the http calls

Reply:
Now that I think about it. The result can be in the session state (since it will have to be in state somewhere if we display it), we should just clear it after the user closes the dialog or go to a different page.

I think having the encryption and decryption function calls along will all the server calls in vuex makes sense from a structure and auditability perspective.

Reply:
No need to save it if we clear it when the dialog box closes unless we allow it to be seen multiple times.
We can trigger the dialog on decryption success without store the result.

What's your thought?

Reply:
Well, if we do it through vuex actions, we will need to modify the state to be able to display the results, since, afaik, vuex actions should not return a result. Whether we store it inside of the state of the dialog (as in the component's state, not the app state), or the app state does not matter for this, as long as we clear the vuex state when the dialog closes.

I think it makes sense to be explicit about the state in this case. So do the call -> update values in vuex -> display dialog -> close dialog -> erase values from vuex.